src/urllib3/poolmanager.py

```diff
@@ -537,7 +543,7 @@ class ProxyManager(PoolManager):
 
     def __init__(
         self,
-        proxy_url: str,
+        proxy_url: Union[ConnectionPool, str],
```

Comment:
This should be `HTTPConnectionPool` since we use `.scheme` below.

Reply:
Then we have an error in `test_oldapi`
https://github.com/urllib3/urllib3/blob/7c9ca88b7dd5fe6b0e9fac52f72202edd74bd8b2/test/with_dummyserver/test_proxy_poolmanager.py#L165

because `connection_from_url` returns `ConnectionPool` and `ProxyManager` accepts `HTTPConnectionPool`

```
test/with_dummyserver/test_proxy_poolmanager.py:168: error: Argument 1 to "ProxyManager" has incompatible type "ConnectionPool"; expected "Union[HTTPConnectionPool, str]"  [arg-type]
```

Reply:
Bleh, `ConnectionPool` is both useless and getting in the way. Makes me want to make it an empty class and move everything to `HTTPConnectionPool`.

Reply:
Shall we change it to `HTTPConnectionPool` and ignore the mypy error in test?

Reply:
Change both here and `connection_from_url` to be `HTTPConnectionPool`.

Reply:
Done!